.github/workflows/pull-request.yml

```diff
@@ -27,81 +27,48 @@ env:
 
 jobs:
 
-  supported-reqs:
+  build-basic:
 
-    name: 'Supported scripts/requirements.txt'
+    name: "build with requirements.txt"
     runs-on: ubuntu-20.04
 
     steps:
       - uses: actions/checkout@v2
 
-      - name: apt-get install
-        run: sudo apt-get -y install $ubuntu_base_deps
-
-      - name: PATH += .local/bin
-        run: echo "$HOME/.local/bin" >> $GITHUB_PATH
-
-      - name: 'pip install -r scripts/requirements.txt'
-        run: pip install -r scripts/requirements.txt
-
-      # No YAML anchors; this is copy/paste
-      - name: configure SOF doc
+      - name: setup env with requirements.txt
         run: |
-          git clone https://github.com/thesofproject/sof
-          cmake -GNinja -S sof/doc -B _build_doxy
+          sudo apt-get -y install $ubuntu_base_deps
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
+          pip install -r scripts/requirements.txt
 
-      # TODO: change the (bad) default value for SOF_DOC_BUILD in
-      # sof-docs/Makefile and remove this command line override
-      - name: build
-        run: |
-          make html VERBOSE=1 SOF_DOC_BUILD=_build_doxy
-          du -shc _build*/*
+      - name: build doc
+        run: ./.github/actions/build-sof-doc.sh
 
-  lax:
+  build-lax:
 
-    name: "PIP_IGNORE_INSTALLED=0 requirements-lax.txt"
+    name: "build with requirements-lax.txt"
     runs-on: ubuntu-20.04
     # Makefile downgrades the Sphinx warnings, they are not errors any more
     env: {LAX: 1}
     steps:
       - uses: actions/checkout@v2
 
-      - name: apt-get install base dependencies
-        run: sudo apt-get -y install $ubuntu_base_deps
-
-      - name: temporary python3-pil HACK because github is out of date
-        run: sudo apt-get -y install libimagequant0 libwebpdemux2 &&
-           wget http://security.ubuntu.com/ubuntu/ubuntu/pool/main/p/pillow/python3-pil_7.0.0-4ubuntu0.2_amd64.deb &&
-           sudo dpkg -i python3-pil_*.deb
-
-      - name: apt-get instead of pip
-        run: sudo apt-get -y install
-               python3-pip python3-setuptools python3-wheel
-               python3-sphinx python3-breathe python3-docutils
-               python3-sphinx-rtd-theme python3-sphinxcontrib.plantuml
-
-      - name: PATH += .local/bin
-        run: echo "$HOME/.local/bin" >> $GITHUB_PATH
-
-      # should be a no-op
-      - name: 'pip install -r scripts/requirements-lax.txt'
-        run: pip install -r scripts/requirements-lax.txt
-
-      - name: config tweaks
+      - name: setup env with requirements-lax.txt
         run: |
-           # sphinx-build < 1.7 doesn't support "auto" and we're not sure
-           # all default modules are "thread-safe"
-           sed -i -e '/SPHINXBUILD/ s/-j *auto/-j 1/' Makefile
-           # We don't want plantUML to raise the contribution bar
-           sed -i -e 's/^\(plantuml_output_format *=\).*/\1 "none"/' conf.py
+          sudo apt-get update
```

Comment:
As already mentioned in #321, if we add `apt-get update` here then this job will be slow forever. The `wget` is slow too but it will hopefully fail one day in the future. When it starts failing we remove it and we are fast again. I should have explained this in a comment.

Reply:
This looks like a one time task ? Can we optimise this later for only running once ?

Reply:
I just want to reproduce what we will do locally from the document we shared. But it seems I still get confused about how to use it.

Reply:
> If we add apt-get update here then this job will be slow forever. The wget is slow too but it will hopefully fail one day in the future. When it starts failing we remove it and we are fast again

I was wrong, Github Actions require `apt-get update` before `apt-get install` because unlike normal Ubuntu systems they don't `apt-get update` in the background. Longer story in https://github.com/actions/virtual-environments/issues/ 1757 (whitespace inserted to avoid spamming 1757)